Webpack 5 Throws ModuleNotFoundError

Hi, I'm using Webpack^5.46.0, @storybook/react^6.3.4, msw-storybook-addon^1.2.0, and msw^0.35.0. I have Storybook configured to use Webpack 5. Whenever the Storybook dev server is running and I modify the preview to initialize the worker, it works. But when I shutdown the server and restart it, it breaks with the following error:

```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it
```

This error is thrown multiple times for multiple packages. Has this been reported before? Any possible fix?

I just upgraded my project from gatsby 2 to 3, which switches to webpack 5.

I followed all the instructions in the storybook webpack 5 documentation, and Shilman helped me debug it and this error kept happening.

I noticed the `@mswjs` in the ModuleNotFoundError, commented out the msw initialization code in preview.js
```
//import {initialize, mswDecorator} from 'msw-storybook-addon';
//initialize();
//addDecorator(mswDecorator);
```
Storybook launched without issue.

I followed @mouhannad-sh instructions, uncommented the msw storybook code and everything works again.

Here is the stack trace (it repeats this error multiple times, so I only pasted the first occurance).

```
manager (webpack 5.52.1) compiled successfully in 13889 ms
99% done plugins webpack-hot-middlewarewebpack built preview 34cdf0e9d905502cac0d in 19490ms
ModuleNotFoundError: Module not found: Error: Can't resolve 'http' in '/Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/interceptors/ClientRequest'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "http": require.resolve("stream-http") }'
	- install 'stream-http'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "http": false }
    at /Users/steven/Development/project/node_modules/webpack/lib/Compilation.js:1773:28
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:814:13
    at eval (eval at create (/Users/steven/Development/project/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:286:22
    at eval (eval at create (/Users/steven/Development/project/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:442:22
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:124:11
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:676:25
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:885:8
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:1005:5
    at /Users/steven/Development/project/node_modules/neo-async/async.js:6883:13
    at /Users/steven/Development/project/node_modules/webpack/lib/NormalModuleFactory.js:988:45
    at finishWithoutResolve (/Users/steven/Development/project/node_modules/enhanced-resolve/lib/Resolver.js:296:11)
    at /Users/steven/Development/project/node_modules/enhanced-resolve/lib/Resolver.js:362:15
    at /Users/steven/Development/project/node_modules/enhanced-resolve/lib/Resolver.js:410:5
    at eval (eval at create (/Users/steven/Development/project/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
resolve 'http' in '/Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/interceptors/ClientRequest'
  Parsed request is a module
  using description file: /Users/steven/Development/project/node_modules/@mswjs/interceptors/package.json (relative path: ./lib/interceptors/ClientRequest)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /Users/steven/Development/project/src
        single file module
          using description file: /Users/steven/Development/project/package.json (relative path: ./src/http)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/src/http.cjs doesn't exist
        /Users/steven/Development/project/src/http doesn't exist
      /Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/interceptors/ClientRequest/node_modules doesn't exist or is not a directory
      /Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/interceptors/node_modules doesn't exist or is not a directory
      /Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/node_modules doesn't exist or is not a directory
      looking for modules in /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules
        single file module
          using description file: /Users/steven/Development/project/node_modules/@mswjs/interceptors/package.json (relative path: ./node_modules/http)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http.cjs doesn't exist
        /Users/steven/Development/project/node_modules/@mswjs/interceptors/node_modules/http doesn't exist
      /Users/steven/Development/project/node_modules/@mswjs/node_modules doesn't exist or is not a directory
      /Users/steven/Development/project/node_modules/node_modules doesn't exist or is not a directory
      looking for modules in /Users/steven/Development/project/node_modules
        single file module
          using description file: /Users/steven/Development/project/package.json (relative path: ./node_modules/http)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/node_modules/http.cjs doesn't exist
        /Users/steven/Development/project/node_modules/http doesn't exist
      /Users/steven/Development/amex/node_modules doesn't exist or is not a directory
      /Users/steven/Development/node_modules doesn't exist or is not a directory
      /Users/steven/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/steven/Development/project/test
        single file module
          using description file: /Users/steven/Development/project/package.json (relative path: ./test/http)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/steven/Development/project/test/http.cjs doesn't exist
        /Users/steven/Development/project/test/http doesn't exist
ModuleNotFoundError: Module not found: Error: Can't resolve 'http' in '/Users/steven/Development/project/node_modules/@mswjs/interceptors/lib/interceptors/ClientRequest'
```

--- a/packages/msw-addon/src/mswDecorator.js
+++ b/packages/msw-addon/src/mswDecorator.js
@@ -17,6 +17,13 @@ export function initialize(options) {
     worker.start(options)
     api = worker
   } else {
+    /**
+     * Webpack 5 does not provide node polyfills as it did before.
+     * Also, it can't tell whether a code will be executed at runtime, so it has to process everything. This branch of the conditional statement will NEVER run in the browser, but Webpack can't know so it breaks builds unless we start providing node polyfills.
+     * 
+     * As a workaround, we use __non_webpack_require__ to tell Webpack to ignore this, and we define it to globalThis so it works correctly when running in node.
+     * See https://github.com/webpack/webpack/issues/8826#issuecomment-660594260
+     */
     globalThis.__non_webpack_require__ = require
     const { setupServer } = __non_webpack_require__('msw/node')
     const server = setupServer()

